kvmagent/kvmagent/plugins/vms/vm_host_file_monitor.py

```
@@ -0,0 +1,246 @@
+KVM_REPORT_VM_HOST_FILE_CHANGED = '/kvm/reporthostfilechanged'
+
+# Number of consecutive scan misses before removing a VM from the watch list.
+MISS_COUNT_THRESHOLD = 10
+
+
+def _file_md5(path):
+    # type: (str) -> str | None
+    """Return hex md5 digest of *path*, or None if file does not exist."""
+    if not os.path.isfile(path):
+        return None
+    md5 = hashlib.md5()
```

Ruff (0.15.17)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)